candy-machine/program/src/utils.rs

```diff
@@ -1,55 +1,54 @@
-use anchor_lang::prelude::{Signer, Sysvar};
-use solana_program::program::invoke;
-use solana_program::system_instruction;
-
-use {
-    crate::{CandyMachine, ErrorCode},
-    anchor_lang::{
-        prelude::{Account, AccountInfo, Clock, ProgramError, ProgramResult, Pubkey},
-        solana_program::{
-            msg,
-            program::invoke_signed,
-            program_pack::{IsInitialized, Pack},
-        },
-    },
-    spl_associated_token_account::get_associated_token_address,
+use crate::{CandyError, CandyMachine};
+use anchor_lang::prelude::*;
+use solana_program::{
+    account_info::AccountInfo,
+    clock::Clock,
+    program::{invoke, invoke_signed},
+    program_memory::sol_memcmp,
+    program_pack::{IsInitialized, Pack},
+    pubkey::{Pubkey, PUBKEY_BYTES},
+    system_instruction,
 };
+use spl_associated_token_account::get_associated_token_address;
+use std::str::from_utf8_unchecked;
 
-pub fn assert_initialized<T: Pack + IsInitialized>(
-    account_info: &AccountInfo,
-) -> Result<T, ProgramError> {
+pub fn assert_initialized<T: Pack + IsInitialized>(account_info: &AccountInfo) -> Result<T> {
     let account: T = T::unpack_unchecked(&account_info.data.borrow())?;
     if !account.is_initialized() {
-        Err(ErrorCode::Uninitialized.into())
+        Err(CandyError::Uninitialized.into())
     } else {
         Ok(account)
     }
 }
 
+pub fn cmp_pubkeys(a: &Pubkey, b: &Pubkey) -> bool {
```

Comment:
We need a centralized crate for these helpers. This is so prone to error since it's dup'ed across every workspace.

Reply:
Agreed. We need a utils lib or something.